src/db/AppleRepository.ts

```diff
@@ -0,0 +1,89 @@
+import e from 'express'
+import { Pool } from 'mysql2/promise'
+import {
+    AppleEpisodePayload,
+    AppleEpisodePlayCountPayload,
+} from '../types/connector'
+
+class AppleRepository {
+    pool
+
+    constructor(pool: Pool) {
+        this.pool = pool
+    }
+
+    // store play counts that is send from the episodes api together with  metadata
+    async storeEpisodesPlayCount(
+        accountId: number,
+        episodesPlayCounts: AppleEpisodePlayCountPayload[][]
+    ): Promise<any> {
+        const replaceStmt = `REPLACE INTO appleEpisodePlayCounts (
+            account_id,
+            episode_id,
+            apc_playscount,
+            apc_totaltimelistened,
+            apc_uniqueengagedlistenerscount,
+            apc_uniquelistenerscount
+            ) VALUES
+            (?,?,?,?,?,?)`
+
+        return await Promise.all(
+            episodesPlayCounts.map(
+                async (
+                    entries: AppleEpisodePlayCountPayload[]
+                ): Promise<any> => {
+                    // the apple api allows an array of playcount objects which doesn't make a lot of sense
+                    // as there is always just one entry returned. therefore, we just use the first and only
+                    // element of the given array of playcount objects
+                    const entry = entries[0]
```

Comment:
We could add a check later to make sure `length == 1` and if not return an error. Not urgent, though.